src/node_crypto.cc

```diff
@@ -4878,6 +4878,26 @@ void GetHashes(const FunctionCallbackInfo<Value>& args) {
 }
 
 
+void GetCurves(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  Local<Array> arr = Array::New(env->isolate());
+  size_t len = EC_get_builtin_curves(NULL, 0);
+  EC_builtin_curve* curves =
+    reinterpret_cast<EC_builtin_curve*>(malloc(sizeof(EC_builtin_curve) * len));
+
+  if (curves) {
```

Comment:
Won't this just fail silently if you run out of memory in the `malloc` call? Perhaps something like [this](https://github.com/nodejs/io.js/blob/ff39ecb91403093c471cce2bb1c1bad34c2a79a6/src/node_crypto.cc#L4277) would be simpler.